src/lib/ext-activitylog.js

```diff
@@ -4,25 +4,51 @@ class Model {
   constructor() {
     this.logs = [];
     this.filter = {
-      id: [],
-      viewType: [],
-      type: [],
+      id: new Set(),
+      viewType: new Set(),
+      type: new Set(),
+      keyword: '',
     };
   }
 
   addNewLogs(logs) {
     this.logs.push(...logs);
   }
 
-  addFilter({ logKey, valueEquals }) {
-    this.filter[logKey].push(valueEquals);
+  /**
+   * Sets the value of any key in filter
+   * @param {string} logKey - The key in the filter.
+   * @param {Set} filterDescriptor - It contains collection of string for any key except keyword key.
+   * @param {string} filterDescriptor - It contains string for keyword key.
+   */
+  setFilter({ logKey, filterDescriptor }) {
+    this.filter[logKey] = filterDescriptor;
   }
```

Comment:
@atiqueahmedziad 
This jsdoc looks still wrong to me (e.g. there are two different definition of the filterDescriptor property, and if I would be a developer that is looking to this jsdoc to understand this function I would still be confused on how I should use it) and the current method signature does confuse me a bit (I had to follow the code to actually understand how that was going to be used).

In my opinion this method could take as its parameter an object with "the subset of the same properties of the filter object" that the caller want to update (with the exact same properties of the filter object as defined in the constructor al line 6, all properties would be optional to allow the caller to update a particular part of the filter and leave the rest unmodified).

I think that this would make the method easier to use and it should also make it easier to express it in the jsdoc in a non confusing way.

The method signature would be something like
```
setFilter(updatedFilterProps) {
  // here the code to merge updateFilterProps into this.filter
}
```

Let me know what do you think (and if my comment is clear enough)

Reply:
Thanks @rpl for the idea. I have implemented this. :-)